src/dc_chat.rs

```diff
@@ -1726,10 +1716,10 @@ pub unsafe fn dc_remove_contact_from_chat(
                 );
             } else {
                 /* we should respect this - whatever we send to the group, it gets discarded anyway! */
-                if !contact.is_null() {
+                if let Ok(contact) = Contact::get_by_id(context, contact_id) {
                     if (*chat).param.get_int(Param::Unpromoted).unwrap_or_default() == 0 {
                         (*msg).type_0 = Viewtype::Text;
-                        if (*contact).id == 1 as libc::c_uint {
+                        if contact.id == 1 as libc::c_uint {
```

Comment:
I think 1 here is probably `contants::DC_CONTACT_ID_SELF`, good opportunity to fix that.